src/plugins/discover/public/application/components/discover_grid/get_render_cell_value.tsx

```diff
@@ -60,6 +62,12 @@ export const getRenderCellValueFn = (
     return <span>-</span>;
   }
 
+  const { fields } = row;
+  let fieldsToShow = fields || [];
+  if (fields) {
+    fieldsToShow = getFieldsToShow(Object.keys(fields), indexPattern, showMultiFields);
```

Comment:
same like in my previous comment, fieldsToShow shouldn't be calculated by row, since it doesn't change and with lots of fields, this might have an impact to performance